### PR TITLE
user email is now optional

### PR DESCRIPTION
**Pull request by ericbuckley**
The prominent change is making email optional in the `UserIdentity` class, this will allow access for inbound SSOe users that do not have an email address on file.

There are a few cases in the vets-api where `user.email` is expected to always have a value.  To remedey this we've introduced a new `common_name` attribute that will be set to the `va_eauth_commonname` value when authenticated via SSOe and the user's email when not using SSOe.

In the cases where an email address is optional, we've updated the code to use `user.email` or an empty string.  In the cases where its required, we've found that the caller just needs a unique identifier, so we've updated those cases to use the new `common_name` attribute.

closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/8305

### CODE REVIEW

**Comment by annaswims** [comment]
I tink there are a lot of places that we're calling user.email, assuming that we'll get a value.  Would it make sense  for user.email to fall back to the email address in vets360/VA profile if we don't get one from the identity provider?

**Comment by Samara-Strauss** [comment]
> I think there are a lot of places that we're calling user.email, assuming that we'll get a value. Would it make sense for user.email to fall back to the email address in vets360/VA profile if we don't get one from the identity provider? Maybe @Samara-Strauss would have thoughts on that.

I want to say yes to this, but I guess it depends on the reason/use case(s).

**Comment by ericbuckley** [comment]
@annaswims this is still a work in progress, and I'll make sure to update the PR with a description before moving it to review, but here is the original issue that @patrickvinograd created to outline where in vets-website/vets-api the users email address was being used https://github.com/department-of-veterans-affairs/va.gov-team/issues/8305

**Comment by patrickvinograd** [comment]
Missing the use in `/lib/evss/jwt.rb`. Guidance from EVSS team was to omit the `email` claim from the JWT rather than supplying an empty value.

**Comment by ericbuckley** [comment]
> Missing the use in `/lib/evss/jwt.rb`. Guidance from EVSS team was to omit the `email` claim from the JWT rather than supplying an empty value.

As the code stands it should be submitting a null value, `{"email": null, ....}`.  But you're saying we need to remove the attribute altogether, correct?

**Comment by kfrz** [comment]
This looks good as far as I can tell, only thing I didn't see addressed here is this use case:

> Form526OptInsController - accepts email as input (unknown if prefilled on frontend)

Did you verify this isn't prefilled on frontend? 

No explicit test cases need to be added?

**Comment by ericbuckley** [comment]
> This looks good as far as I can tell, only thing I didn't see addressed here is this use case:
> 
> > Form526OptInsController - accepts email as input (unknown if prefilled on frontend)
> 
> Did you verify this isn't prefilled on frontend?
> 
> No explicit test cases need to be added?

@kfrz just confirmed that it the email in the form is initialized to the empty string and not any email value from the user's profile https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/disability-benefits/526EZ/components/create526EmailForm.jsx#L16

**Comment by ericbuckley** [comment]
@patrickvinograd do you want to see any more changes to this PR before I merge?

File: lib/evss/jwt.rb [hunk 1]
```diff
@@ -39,7 +38,8 @@ def payload
         prefix: '', # TODO: i thought we had these in mvi...
         suffix: '',
         correlationIds: @user.mvi&.profile&.full_mvi_ids
-      }
+        # only include the email address if its not nil
+      }.merge({ email: @user.email }.compact)
```
Comment (patrickvinograd) [inline lib/evss/jwt.rb:42]: 👍

**Pull request merged**